src/nlp.jl

```diff
@@ -71,7 +79,7 @@ function KnitroSolver(
     pop!(kwargs, :z0)
   end
 
-  jrows, jcols = m > 0 ? jac_structure(nlp) : (Int[], Int[])
+  jrows, jcols = nlp.meta.nnln > 0 ? jac_nln_structure(nlp) : (Int[], Int[])
```

Comment:
Doesn't `jac_nln_structure()` return empty arrays if `nlp.meta.nnln == 0` ?

Reply:
Not necessarily, it may return an error NotImplemented.
For instance, for models with linear constraints only, we generally don't implement the nonlinear functions.